langchain4j-core/src/main/java/dev/langchain4j/model/input/DefaultPromptTemplateFactory.java

```diff
@@ -63,7 +64,10 @@ private static String replaceAll(String template, String variable, Object value)
             if (value == null || value.toString() == null) {
                 throw illegalArgument("Value for the variable '%s' is null", variable);
             }
-            return template.replace(inDoubleCurlyBrackets(variable), value.toString());
+            String replacement = value instanceof Object[]
```

Comment:
I am not sure if it should be fixed here.
If you check `Chef.createRecipeFrom(String... ingredients)`, it worked fine.
Please check `DefaultAiServices.findTemplateVariables`, it works differently for `Chef.createRecipeFrom` and `Chef.createRecipeFromVarargs`

Reply:
@langchain4j the problem only happens if you have more than 1 parameter, check the example I gave in the `change` section of the PR descriptions